### PR TITLE
doc: releases: Enforce linking to Pull Requests in migration guide

### PR DESCRIPTION
**Pull request by carlescufi**
In order to provide additional context and information to Zephyr users,
    ensure that each entry in the migration guide contains a link to the
    GitHub Pull Request that introduced the corresponding change.

### CODE REVIEW

**Review by fabiobaltieri — DISMISSED** [review]
good idea

File: doc/releases/migration-guide-3.7.rst [hunk 1]
```diff
@@ -20,7 +20,7 @@ Build System
 
 * Completely overhauled the way SoCs and boards are defined. This requires all
   out-of-tree SoCs and boards to be ported to the new model. See the
-  :ref:`hw_model_v2` for more detailed information.
+  :ref:`hw_model_v2` for more detailed information. (:github:`69607`)
```
Comment (RichardSWheatley) [inline doc/releases/migration-guide-3.7.rst:23]: @carlescufi is the (:github:'69607') Intended to add a link to a PR? or Issue?
Just a question as I am looking at doing something similar outside of Zephyr.

**Review by henrikbrixandersen — COMMENTED** [review]
This doesn't enforce anything, though.

It puts the work of ensuring each entry contains a link to a PR on the shoulders of the reviewers.

I tried to insist people do this at the start of the v3.6 release cycle, but gave up since it's quite difficult to manage without an automated check.

**Comment by RichardSWheatley** [comment]
> This doesn't enforce anything, though.
> 
> It puts the work of ensuring each entry contains a link to a PR on the shoulders of the reviewers.
> 
> I tried to insist people do this at the start of the v3.6 release cycle, but gave up since it's quite difficult to manage without an automated check. 

I tried viewing the rst file in the source directory and the PR did not contain a link but just said the number. How should I view it to see the link?

**Comment by henrikbrixandersen** [comment]
> I tried viewing the rst file in the source directory and the PR did not contain a link but just said the number. How should I view it to see the link?

You need to look at the rendered output from this PR by following the link in the Documentation Build CI check: https://docs.zephyrproject.org/latest/index.html

File: doc/releases/migration-guide-3.7.rst [hunk 2]
```diff
@@ -396,19 +397,19 @@ Userspace
...
     are deprecated. Use :kconfig:option:`CONFIG_X86_DISABLE_SSBD` and
-    :kconfig:option:`CONFIG_X86_ENABLE_EXTENDED_IBRS` instead.
+    :kconfig:option:`CONFIG_X86_ENABLE_EXTENDED_IBRS` instead. (:github:`69690`)
 
 * POSIX arch:
 
   * LLVM fuzzing support has been refactored. A test application now needs to provide its own
     ``LLVMFuzzerTestOneInput()`` hook instead of relying on a board provided one. Check
-    ``samples/subsys/debug/fuzz/`` for an example.
+    ``samples/subsys/debug/fuzz/`` for an example. (:github:`69690`)
```
Comment (aescolar) [inline doc/releases/migration-guide-3.7.rst:412]: 71378

File: doc/releases/migration-guide-3.7.rst [hunk 3]
```diff
@@ -374,7 +375,7 @@ Modem
 =====
 
 * The ``CONFIG_MODEM_CHAT_LOG_BUFFER`` Kconfig option was
-  renamed to :kconfig:option:`MODEM_CHAT_LOG_BUFFER_SIZE`.
+  renamed to :kconfig:option:`MODEM_CHAT_LOG_BUFFER_SIZE`. (:github:`70405`)
```
Comment (aescolar) [inline doc/releases/migration-guide-3.7.rst:378]: This is really tangentional but if you are changing this line can you fix this:
```suggestion
  renamed to :kconfig:option:`CONFIG_MODEM_CHAT_LOG_BUFFER_SIZE`. (:github:`70405`)
```

File: doc/releases/migration-guide-3.7.rst [hunk 4]
```diff
@@ -106,16 +106,16 @@ Device Drivers and Devicetree
...
   The following types have been renamed:
 
   * ``emul_sensor_backend_api`` to :c:struct:`emul_sensor_driver_api`
   * ``emul_bbram_backend_api`` to :c:struct:`emul_bbram_driver_api`
   * ``usbc_ppc_drv`` to :c:struct:`usbc_ppc_driver_api`
 
 * The driver for :dtcompatible:`maxim,max31790` got split up into a MFD and an
-  actual PWM driver. Previously, an instance of this device could have been
-  defined like this:
+  actual PWM driver. (:github:`71097`)
```
Comment (aescolar) [inline doc/releases/migration-guide-3.7.rst:117]: 68433

**Review by aescolar — CHANGES_REQUESTED** [review]

**Comment by carlescufi** [comment]
> This doesn't enforce anything, though.

It documents that it is required to include the link, it's a first start.

> It puts the work of ensuring each entry contains a link to a PR on the shoulders of the reviewers.

Agreed, we will take a look at a check later.

**Review by aescolar — APPROVED** [review]

**Review by kartben — APPROVED** [review]
I thought we had documented this already when we discussed introducing this change, so thanks for actually taking care of it! :)

**Review by henrikbrixandersen — COMMENTED** [review]
While I think introducing this requirement for migration guide entries is a good idea, I fear introducing it without any checks in place will place the burden on the reviewers and release managers.

**Comment by fabiobaltieri** [comment]
> While I think introducing this requirement for migration guide entries is a good idea, I fear introducing it without any checks in place will place the burden on the reviewers and release managers.

Yeah at the scale at which the project operates and with many people rotating through the roles it would be a good idea if any changes like this one were introduced directly as a compliance check rather than (just) documentation. In this state you may as well consider it just a guideline.

**Comment by fabiobaltieri** [comment]
So actually the PR title is misleading. It's not really enforcing anything.

**Comment by carlescufi** [comment]
> So actually the PR title is misleading. It's not really enforcing anything.

The documentation uses the verb "must", that's "enforcing" in this context.

**Pull request merged**